### PR TITLE
PMI-97: Convert and parse INTERVALS.

### PR DESCRIPTION
**Pull request by redcatbear**
Summary:
- PMI-97: Convert INTERVALS to milliseconds and vice-versa.

### CODE REVIEW

File: doc/system_requirements.md [hunk 1]
```diff
@@ -247,19 +254,19 @@ Covers:
...
 
----
+ESB converts values of type `INTERVAL` to integer and vice-versa.
 
-SELECT
-* Fields
-* Asterisk ("*")
+Rationale:
+
+Neighboring systems of an Exasol database often do to have equivalent data types, so conversion to a primitive data type is required.
```
Comment (bobkodex) [inline doc/system_requirements.md]: > ... do to have equivalent
This section is unclear

File: src/main/java/com/exasol/datatype/interval/IntervalDayToSecond.java [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+ * </p>
+ * <ul>
+ * <li>days</li>
+ * <li>hours</li>
+ * <li>minutes</li>
+ * <li>seconds</li>
+ * <li>milliseconds (or fraction of seconds)</li>
+ * </ul>
+ *
+ * Since milliseconds are the highest resolution, each interval can also be expresses as a total number of milliseconds.
```
Comment (bobkodex) [inline src/main/java/com/exasol/datatype/interval/IntervalDayToSecond.java]: typo: 
>can be expressed

File: doc/design.md [hunk 3]
```diff
@@ -112,4 +112,38 @@ Covers:
...
+* `req~integer-interval-conversion~1`
+
+Needs: impl, utest
+
+#### Parsing INTERVAL DAY TO SECOND From Strings
+`dsn~exasol.parsing-interval-day-to-second-from-strings~1`
+
+The data converter can parse `INTERVAL DAY TO SECOND` from strings in the following format:
+
+    inverval-d2s = [ days SP ] hours ":" minutes [ ":" seconds [ "." milliseconds ] ]
```
Comment (bobkodex) [inline doc/design.md]: typo: inverval-d2s --> interval-d2s
At first glance, notation is not completely clear.
Maybe concret examples for string intervals (interval-d2s, hours, minutes ...) could be helpul (just suggestion)

File: doc/system_requirements.md (hunk 1 above)
Comment (redcatbear) [reply doc/system_requirements.md]: Fixed:
> "... do **not** have ..."

File: doc/design.md (hunk 3 above)
Comment (redcatbear) [reply doc/design.md]: Added examples. Will add introduction including ABNF mention in separate ticket.

**Pull request merged**